### PR TITLE
Foreground init

### PR DESCRIPTION
**Pull request by makuk66**

### CODE REVIEW

**Comment by tianon** [comment]
This is better, but still suffers from the problem that all the `initdb` scripts have no way other than a background process that waits of actually connecting to the running database reliably (and if one of them fails, the process doesn't fail).  I'm also not sure I completely understand the use case for the `solr-demo`, etc. shortcuts -- can you elaborate on how those are expected to be used a little bit?  (Since it seems to me that it'd be a more straightforward experience for users to either provide a simple `initdb` script to perform some of those actions or to start up the main database and perform those from a separate container.)

**Comment by makuk66** [comment]
> all the initdb scripts have no way other than a background process that waits of actually connecting to the running database reliably (and if one of them fails, the process doesn't fail). 

An initdb script can make changes to the disk configuration before Solr gets started as normal.
An initdb script can start Solr in the background, talk to it from the foreground, do whatever error-checking and reporting it wants (and exit if needed), and then stop it, ready to get started again in the foreground as normal. Both satisfy your criteria as I understand them. I longer have any documentation or examples of scripts that run in the background while Solr runs in the foreground. What am I missing?

> I'm also not sure I completely understand the use case for the solr-demo, etc. shortcuts -- can you elaborate on how those are expected to be used a little bit? (Since it seems to me that it'd be a more straightforward experience for users to either provide a simple initdb script to perform some of those actions or to start up the main database and perform those from a separate container.)

See https://github.com/docker-solr/docker-solr/issues/14. To do anything interesting with Solr, you need to create a index. You cannot create a Solr core solely through the UI or API at this time; some on-disk work is required. So it can't be done from another container. Solr provides a command-line utility to do this, but that has to run on the same container, so currently requires `docker exec` (which is what we used to recommend), or writing external scripts mapped into the container (as a volume) or from a custom Dockerfile, or writing long command-lines. That is inconvenient and error-prone. Using the `solr-create` and `solr-precreate` commands in the docker entrypoint you can do it simply in a single `docker run`, and have a container ready to use from the UI or remotely through the API from some other container.

The `solr-demo` is simply an extension of the above that loads some of Solr's provided example data into a core, so you can start showing off Solr searching with a single command. You could do that remotely from a second container (if you start the first one with a `solr-create` or `solr-precreate`), or you could do it from an initdb script. But surely that is less straightforward.

File: scripts/wait-for-solr.sh [hunk 1]
```diff
@@ -1,8 +1,7 @@
 #!/bin/bash
 #
 # A helper script to wait for solr
-set -x
-until $(wget -O - http://localhost:8983 | grep -q -i solr); do
+until $(wget -q -O - http://localhost:8983 | grep -q -i solr); do
```
Comment (tianon) [inline scripts/wait-for-solr.sh]: This is currently taking the output of `wget | grep -q` and trying to run it as a command, so can be replaced with simply:
```bash
until wget -q -O - http://localhost:8983 | grep -q -i solr; do
```

**Comment by tianon** [comment]
Ok, fair enough -- thanks for taking the time to explain. :+1:

LGTM (although one minor comment on the `wait-for-solr.sh` script)

**Pull request merged**